### PR TITLE
Add molecule tests to collection

### LINKED ISSUE #41: Add basic tests
We made good progress with other roles by adding tests as early as possible. Since this is no role but a collection, we need to change some of the process.

An example on how to check collections with molecule can be found here: https://www.jeffgeerling.com/blog/2019/how-add-integration-tests-ansible-collection-molecule

This will help with #32 and vice versa.

### PR DESCRIPTION
**Pull request by widhalmt**
Fixes #41

### CODE REVIEW

**Comment by widhalmt** [comment]
Some of the checks won't run locally as easy as they do in GitHub actions. So this might get a long trail of trial&error commits before it's ready to merge.

**Comment by widhalmt** [comment]
@lbetz / @mkayontour please tell me what you think of the code.

Please *don't* just merge it now. We'll need to make *very* sure to have the commits squashed during merging and I want to review all the commit messages. There's a lot in them that was just used "under way" but doesn't bring any benefit when we have it in the history of main.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # ansible-collection-icinga
 
+[![CI](https://github.com/Icinga/ansible-collection-icinga/workflows/CI/badge.svg?event=push)](https://github.com/Icinga/ansible-collection-icinga/actions?query=workflow%3ACI)
```
Comment (mkayontour) [inline README.md:3]: The CI Status doesn't work, this generic one does work. 
https://github.com/Icinga/ansible-collection-icinga/actions/workflows/molecule.yml/badge.svg

File: molecule/default/INSTALL.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
+*******
```
Comment (mkayontour) [inline molecule/default/INSTALL.rst:1]: This file is not required for the test itself, can be removed I guess?

**Review by mkayontour — CHANGES_REQUESTED** [review]
Everything looks good to me, for the basics this is enough. I'll add a few more cases and scenarios after the merge is done!

The install.rst is something I would remove - because it could be distracting but let me hear what you think. 
Thanks buddy :)

File: README.md (hunk 1 above)
Comment (widhalmt) [reply README.md:3]: The other badge should work, once it's merged. The pipelines the link is referring to, does not yet exist. But we can always change it to the more generic one to be safe. I'll change it.

File: molecule/default/INSTALL.rst (hunk 2 above)
Comment (widhalmt) [reply molecule/default/INSTALL.rst:1]: Yes, you're right. I just copied over the default without checking this file. I'll delete it.

**Comment by widhalmt** [comment]
Thanks for your feedback, @mkayontour . I tried to apply the changes you requested. Could you have another look?

**Review by mkayontour — APPROVED** [review]

**Comment by widhalmt** [comment]
Talked to @lbetz . He's swamped with other tasks now but he's ok, with me merging. So: merging. Yay.

**Pull request merged**